### PR TITLE
Make literal end indentation rule correctable

### PR DESCRIPTION
**Pull request by erichoracek**
Follows the same ideas in #2192. Thanks in advance for the review!

### CODE REVIEW

**Comment by marcelofabri** [comment]
@erichoracek The performance difference on oss-check is really standing out. Can you run it locally just to make sure it's not a CI instability?

**Comment by erichoracek** [comment]
@marcelofabri Good catch! Looks like I had accidentally made this rule O(N^2) during my refactor, just pushed up a fix 😅

File: CHANGELOG.md [hunk 1]
```diff
@@ -33,6 +33,9 @@
 * Updates the `closure_end_indentation` rule to support autocorrection.  
   [Eric Horacek](https://github.com/erichoracek)
 
+* Updates the `literal_expression_end_indentation` rule to support autocorrection.
```
[the remark this answers is not in the document]
Comment (erichoracek) [reply CHANGELOG.md]: We need a `ChangelogLint` rule that supports autocorrection 😅

[the remark this answers is not in the document]
Comment (marcelofabri) [reply CHANGELOG.md]: We should put this in Danger 🤔

**Comment by erichoracek** [comment]
@jpsim I think to do so we'd want to introduce some sort of `ASTCorrectible` construct, which (IMHO) feels like part of another pull request. What do you think?

**Comment by jpsim** [comment]
Let's merge this, code looks good as-is, but I do think it'd be worth deduplicating some common code with `closure_end_indentation`.

**Pull request merged**